testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayViewModel.kt

```diff
@@ -30,6 +33,7 @@ class MiniAppDisplayViewModel constructor(
         try {
             _isLoading.postValue(true)
             miniAppDisplay = miniapp.create(appId, versionId)
+            hostLifeCycle?.addObserver(miniAppDisplay)
             _miniAppView.postValue(miniAppDisplay.getMiniAppView())
```

Comment:
I was thinking about passing the `hostLifeCycle` into `miniAppDisplay.getMiniAppView(hostLifeCycle)` to put a constraint to developer.

`RealMiniAppDisplay` could have this function:
 
```
fun registerLifecycle(lifecycle: Lifecycle) {
    lifecycle.addObserver(this)
}
```

Reply:
As we discussed it out, there were a few reasons behind the approach I followed:
- I was intending to not to force the behavior and be neutral for a variety of host app developer needs (the advanced ones could utilize lifecycle and the traditional ones could just do it manually). It may even happen that some of the old apps might not even have a lifecycle in there.

- At this moment, `RealMiniAppDisplay` has no dependency on its operations which is fulfilled via lifecycle and is mandated. it would have definitely made sense to add it to the contract if there was more operational dependency. I think in the near future we might just add it once there is an increase in dependency.